commands/guild/twitch-announcer.js

```diff
@@ -0,0 +1,355 @@
+            );
+            var canvas = Canvas.createCanvas(result.width, result.height);
+            var ctx = canvas.getContext('2d');
+            // Since the image takes time to load, you should await it
+            var background = await Canvas.loadImage(
+              gameInfo.data[0].box_art_url.replace(/-{width}x{height}/g, '')
+            );
+            // This uses the canvas dimensions to stretch the image onto the entire canvas
+            ctx.drawImage(background, 0, 0, canvas.width, canvas.height);
+            // Use helpful Attachment class structure to process the file for you
+            var attachment = new MessageAttachment(
+              canvas.toBuffer(),
+              'box_art.png'
+            );
+          } catch (e) {
+            clearInterval(Ticker);
+            message.say(':x: Twitch Announcer has stopped!\n' + e);
+            return;
+          }
+
+          //Online Embed
+          const onlineEmbed = new MessageEmbed()
+            .setAuthor(
+              `Twitch Announcement: ${user.data[0].display_name} Online!`,
+              user.data[0].profile_image_url,
+              'https://twitch.tv/' + user.data[0].display_name
+            )
+            .setURL('https://twitch.tv/' + user.data[0].display_name)
+            .setTitle(user.data[0].display_name + ' is playing ' + currentGame)
+            .addField('Stream Title:', streamInfo.data[0].title)
+            .addField('Currently Playing:', streamInfo.data[0].game_name, true)
+            .addField('Viewers:', streamInfo.data[0].viewer_count, true)
+            .setColor('#6441A4')
+            .setFooter(
+              'Stream Started',
+              'https://static.twitchcdn.net/assets/favicon-32-d6025c14e900565d6177.png' // Official icon link from Twitch.tv
+            )
+
+            .setImage(
+              streamInfo.data[0].thumbnail_url
+                .replace(/{width}x{height}/g, '1280x720')
+                .concat('?r=' + Math.floor(Math.random() * 10000 + 1)) // to ensure the image updates when refreshed
+            )
+            .setTimestamp(streamInfo.data[0].started_at)
+            .attachFiles(attachment)
+            .setThumbnail('attachment://box_art.png');
+
+          if (user.data[0].broadcaster_type == '')
+            onlineEmbed.addField('Rank:', 'BASE!', true);
+          else {
+            onlineEmbed.addField(
+              'Rank:',
+              user.data[0].broadcaster_type.toUpperCase() + '!',
+              true
+            );
+          }
+
+          //Online Send
+          if (DBInfo.botSay.toLowerCase() != 'none') {
+            await announcedChannel.message.say(DBInfo.botSay),
+              await announcedChannel.send(onlineEmbed);
+            var embedID = announcedChannel.lastMessage.id;
+          } else {
+            await announcedChannel.send(onlineEmbed);
+            var embedID = announcedChannel.lastMessage.id;
```

Comment:
embedID is already defined

Reply:
its in an `Else` only one or the other and, never both if i remove one it will never be defined in the other situation